packages/relay/src/lib/eth.ts

```diff
@@ -504,11 +504,50 @@ export class EthImpl implements Eth {
 
     try {
       let weibars: BigNumber | number = 0;
-
       const mirrorAccount = await this.mirrorNodeClient.getAccount(account, requestId);
-      if (mirrorAccount && mirrorAccount.balance) {
-        weibars = mirrorAccount.balance.balance * constants.TINYBAR_TO_WEIBAR_COEF
-      } else {
+      let balanceFound = false;
+
+      // A blockNumberOrTag has been provided. If it is `latest` or `pending` retrieve the balance from /accounts/{account.id}
+      if (mirrorAccount && !EthImpl.blockTagIsLatestOrPending(blockNumberOrTag)) {
+        const blockNumberHex = EthImpl.numberTo0x(blockNumber);
+        const latestBlockNumber = await this.blockNumber(requestId);
+
+        // If the parsed blockNumber is the same as the one from the latest block retrieve the balance from /accounts/{account.id}
+        if (blockNumberHex !== latestBlockNumber) {
+          const blockPromises = [
```

Comment:
If I understand correctly, currently this code does up to 4 requests for the blocks. Once to eventually translate the tag [here ](https://github.com/hashgraph/hedera-json-rpc-relay/pull/549/files#diff-dd8c7990f7764960d05dcb4c1b5ae7eee48b425b1b83d6dcc7e7ccc442330ef0R497), once to get the latest block number [here](https://github.com/hashgraph/hedera-json-rpc-relay/pull/549/files#diff-dd8c7990f7764960d05dcb4c1b5ae7eee48b425b1b83d6dcc7e7ccc442330ef0R513) and 2 more here, if the block number and the lates block number are the same. Can't we just use `getHistoricalBlockResponse` for both the from he start and skip the previous 2 requests. This way we will have both the block numbers and the blocks data with just 2 requests. Also shouldn't this [cached request](https://github.com/hashgraph/hedera-json-rpc-relay/pull/549/files#diff-dd8c7990f7764960d05dcb4c1b5ae7eee48b425b1b83d6dcc7e7ccc442330ef0R499-R503) be performed before the once to the mirror node?